src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/ChainPal.cs

```diff
@@ -35,7 +38,319 @@ public static bool ReleaseSafeX509ChainHandle(IntPtr handle)
             TimeSpan timeout,
             bool disableAia)
         {
-            throw new NotImplementedException(nameof(BuildChain));
+            var chainPal = new AndroidCertPath();
```

Comment:
It doesn't look like we do anything with the `disableAia` parameter here. This maps to `X509ChainPolicy.DisableCertificateDownloads` from the public API.

Does Android have a way to prevent fetching intermediate certificates? If not, should a PNSE be thrown if someone sets it to true? Should we ignore it and document that it does not work in Android?

There is some precedent for ignoring it with macOS. In macOS, this property only works if revocation checking is disabled as well.

Reply:
It is actually that we don't have a good way to enable fetching. This is part of my 'not handled in this change'. By default, the fetching is disabled and can only be enabled though a [system property](https://docs.oracle.com/javase/8/docs/technotes/guides/security/certpath/CertPathProgGuide.html#AIA) which gets read and stored once. I don't think we really have a good way of dealing with that - I think we'd have to do something like the Unix/OpenSSl implementation's `CertificateAssetDownloader`.

Reply:
Until the need is proven otherwise, I'm OK with saying/believing that Android's use cases don't require AIA fetching.